src/EditorFeatures/CSharpTest/ConvertProgram/ConvertToTopLevelStatementsAnalyzerTests.cs

```diff
@@ -0,0 +1,1193 @@
+    {
+        System.Console.WriteLine(args);
+    }
+}
+",
+                FixedCode = @"
+int x = 0;
+
+void OtherMethod()
+{
+    return;
+}
+
+System.Console.WriteLine(args);
+",
+                LanguageVersion = LanguageVersion.CSharp9,
+                TestState = { OutputKind = OutputKind.ConsoleApplication },
+                Options = { { CSharpCodeStyleOptions.PreferTopLevelStatements, true, NotificationOption2.Suggestion } },
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task TestNotWithUnsafeMethod()
+        {
+            var code = @"
+class Program
+{
+    private static int x = 0;
+
+    unsafe static void OtherMethod()
+    {
+        return;
+    }
+
+    static void Main(string[] args)
+    {
+        System.Console.WriteLine(args);
+    }
+}
+";
+            await new VerifyCS.Test
+            {
+                TestCode = code,
+                FixedCode = code,
+                LanguageVersion = LanguageVersion.CSharp9,
+                TestState = { OutputKind = OutputKind.ConsoleApplication },
+                Options = { { CSharpCodeStyleOptions.PreferTopLevelStatements, true, NotificationOption2.Suggestion } },
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task TestOtherComplexMethodBecomesLocalFunction()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = @"
+using System.Threading.Tasks;
+
+class Program
+{
+    private static int x = 0;
+
+    static async Task OtherMethod<T>(T param) where T : struct
+    {
+        return;
+    }
+
+    static void {|IDE0210:Main|}(string[] args)
+    {
+        System.Console.WriteLine(args);
+    }
+}
+",
+                FixedCode = @"
+using System.Threading.Tasks;
+
+int x = 0;
+
+async Task OtherMethod<T>(T param) where T : struct
+{
+    return;
+}
+
+System.Console.WriteLine(args);
+",
+                LanguageVersion = LanguageVersion.CSharp9,
+                TestState = { OutputKind = OutputKind.ConsoleApplication },
+                Options = { { CSharpCodeStyleOptions.PreferTopLevelStatements, true, NotificationOption2.Suggestion } },
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task TestAwaitExpression()
+        {
+            await new VerifyCS.Test
+            {
+                TestCode = @"
+using System.Threading.Tasks;
+
+class Program
+{
+    static async Task {|IDE0210:Main|}(string[] args)
+    {
+        await Task.CompletedTask;
+    }
+}
+",
+                FixedCode = @"
+using System.Threading.Tasks;
+
+await Task.CompletedTask;
+",
+                LanguageVersion = LanguageVersion.CSharp9,
+                TestState = { OutputKind = OutputKind.ConsoleApplication },
+                Options = { { CSharpCodeStyleOptions.PreferTopLevelStatements, true, NotificationOption2.Suggestion } },
+            }.RunAsync();
+        }
+
+        [Fact]
+        public async Task TestInNamespaceWithOtherType()
```

Comment:
Perhaps a version of this test with file scoped namespace? Unless I missed it

Reply:
will do!

Reply:
added a bunch of file scoped namespace tests